pep-9999.rst

```diff
@@ -0,0 +1,257 @@
+PEP: 9999
+Title: Package index configuration standardization
+Author: Fredrik Averpil <fredrik at averpil.com>
+Status: Draft
+Type: Informational
+Created: 05-May-2021
+Post-History: <dates of postings to python-ideas and/or python-dev>
+
+
+Abstract
+========
+
+This PEP standardizes configuration of PyPI repositories.
+
+
+Motivation
+==========
+
+Workflow tools come up with their own configuration mechanisms for PyPI repositories, as there is no standard for this. This causes fragmentation and pain points.
+
+Examples of such configuration files:
+
+* Pip's ``pip.conf`` and ``pip.ini``
+* Packaging's ``.pypirc`` (also used by tools like Flit and Twine)
+* Poetry's ``config.toml`` and ``auth.toml``
+
+
+Rationale
+=========
+
+While this PEP aim to come up with a unified PyPI repository configuration, thought has been given so to be able to extend the configuration later. For example, PyPI repositories are assumed in this PEP, but a ``type`` attribute could be introduced which would default to ``pypi`` down the line, to allow for other package repository types. Any tool could also start taking advantage of the fact that the configuration files and their discovery is a standard, and use this for their own configuration.
```

Comment:
Important to note that the only types that are expected are `simple` (legacy), `json` and `links`. I do not think authentication varies beyond username/passord with or without client-certificaties. Outside of this the the only real config one might need is a custome cert support.

Reply:
> @fredrikaverpil added an initial round of thoughts; happy to add some edits over the weekend as well.

Like I mentioned in some replies, would love if you wanted to help out draft this PEP, so feel free to edit away 👍 😄